sklearn/naive_bayes.py

```diff
@@ -1386,6 +1390,8 @@ def _update_cat_count_dims(cat_count, highest_feature):
             return cat_count
 
         def _update_cat_count(X_feature, Y, cat_count, n_classes):
+            if sp.issparse(X_feature):
+                X_feature = X_feature.toarray().flatten()
```

Comment:
If possible, I prefer to avoid calling `toarray` all together and have `_update_cat_count` work on sparse data without the conversion.

**Details**: If `X` started as a CSC, then I think you can extract the feature for a specific column and then create a new `bincount` that works with CSC data. (There are other details for you to work on here, such as the weights and the mask.)

That being said, I am +0.25 on the current solution. It densitfies per column, which is slightly better than densifies all of `X`. A follow up PR can work on making it more efficient.

Reply:
Thanks for the details. I completely agree with you. I don't like calling `toarray` every time either. But I didn't come up with a good idea.

I don't think I fully understand the details, but I will look into it.

Reply:
I understand it now. I'll implement it.